waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
+    cloud = PyntCloud(pd.DataFrame(data))
+    cloud_fpath = f"{parent_path}/{log_id}/lidar/PC_{timestamp}.ply"
+    check_mkdir(str(Path(cloud_fpath).parent))
+    cloud.to_file(cloud_fpath)
+
+
+def dump_object_labels(
+    labels: List[waymo_open_dataset.label_pb2.Label],
+    timestamp: int,
+    log_id: str,
+    parent_path: str,
+    track_id_dict: Dict,
+) -> None:
+    """Saves object labels from Waymo dataset as json files
+
+    Args:
+        labels: A list of Waymo labels
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+        track_id_dict: Dictionary to store object ID to track ID mappings
+    """
+    argoverse_labels = []
+    for label in labels:
+        argoverse_labels.append(build_argo_label(label, timestamp, track_id_dict))
+    json_fpath = f"{parent_path}/{log_id}/per_sweep_annotations_amodal/tracked_object_labels_{timestamp}.json"
+    check_mkdir(str(Path(json_fpath).parent))
+    save_json_dict(json_fpath, argoverse_labels)
+
+
+def build_argo_label(
+    label: waymo_open_dataset.label_pb2.Label, timestamp: int, track_id_dict: Dict
+) -> Dict:
+    """Builds a dictionary that represents an object detection in Argoverse format from a Waymo label
+
+    Args:
+        labels: A Waymo label
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        track_id_dict: Dictionary to store object ID to track ID mappings
+    Returns:
+        label_dict: A dictionary representing the object label in Argoverse format
+    """
+    label_dict = {}
+    label_dict["center"] = {}
+    label_dict["center"]["x"] = label.box.center_x
+    label_dict["center"]["y"] = label.box.center_y
+    label_dict["center"]["z"] = label.box.center_z
+    label_dict["length"] = label.box.length
+    label_dict["width"] = label.box.width
+    label_dict["height"] = label.box.height
+    label_dict["rotation"] = {}
+    qx, qy, qz, qw = yaw_to_quaternion3d(label.box.heading)
+    label_dict["rotation"]["x"] = qx
+    label_dict["rotation"]["y"] = qy
+    label_dict["rotation"]["z"] = qz
+    label_dict["rotation"]["w"] = qw
+    label_dict["label_class"] = LABEL_TYPES[label.type]
+    label_dict["timestamp"] = timestamp
+    if label.id not in track_id_dict.keys():
+        track_id = uuid.uuid4().hex
+        track_id_dict[label.id] = track_id
+    else:
+        track_id = track_id_dict[label.id]
+    label_dict["track_label_uuid"] = track_id
+    return label_dict
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--save-images", default=True, type=bool, help="whether to save images or not"
+    )
+    parser.add_argument(
+        "--save-poses", default=True, type=bool, help="whether to save poses or not"
+    )
+    parser.add_argument(
+        "--save-calibration",
+        default=True,
```

Comment:
in `argparse` the bool type won't allow a False declaration ever (will always be True). Try running this: `python test_argparse_bool.py --save-images False --range-image 2`
```python
import argparse
import pdb

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "--save-images", default=True, type=bool, help="whether to save images or not"
    )

    parser.add_argument(
        "--range-image",
        default=1,
        type=int,
        choices=[1, 2],
        help="which range image to use from Waymo",
    )
    args = parser.parse_args()
    pdb.set_trace()
    main(args)
```
We need to use `store_false` or `store_true` like https://github.com/mseg-dataset/mseg-api/blob/master/mseg/label_preparation/remap_dataset.py#L190

Reply:
Good catch! I used `str2bool` (addressed in [this commit](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/5aa62285b6622c6372a097bc05a24214f4f59757))